libs/langchain/tests/unit_tests/test_cache.py

```diff
@@ -35,33 +40,41 @@ def set_cache_and_teardown(request: FixtureRequest) -> Generator[None, None, Non
     # Will be run before each test
     cache_instance = request.param
     set_llm_cache(cache_instance())
-    if get_llm_cache():
-        get_llm_cache().clear()
+    if llm_cache := get_llm_cache():
+        llm_cache.clear()
     else:
         raise ValueError("Cache not set. This should never happen.")
 
     yield
 
     # Will be run after each test
-    if get_llm_cache():
-        get_llm_cache().clear()
+    if llm_cache:
+        llm_cache.clear()
         set_llm_cache(None)
     else:
         raise ValueError("Cache not set. This should never happen.")
 
 
-def test_llm_caching() -> None:
+async def test_llm_caching() -> None:
     prompt = "How are you?"
     response = "Test response"
     cached_response = "Cached test response"
     llm = FakeListLLM(responses=[response])
-    if get_llm_cache():
-        get_llm_cache().update(
+    if llm_cache := get_llm_cache():
+        # sync test
+        llm_cache.update(
```

Comment:
Not very recommended to call a blocking method in the event loop but I guess for a test it's ok ?